### PR TITLE
fix(driver): fix some fillers while supporting the new testing framework (#1)

### PR DESCRIPTION
**Pull request by Andreagit97**
**What type of PR is this?**

/kind bug

/kind feature

**Any specific area of the project related to this PR?**

/area driver-kmod

/area driver-bpf

**Does this PR require a change in the driver versions?**

No

**What this PR does / why we need it**:

This PR creates some new fillers extracting them from the `autofiller`. One of the most frequent issues is the management of 32 bits signed numbers into 64 bits ones. Almost all syscalls that send a `fd` are affected by this issue, in this PR we address only some of them :) 
To see the complete work take a look here https://github.com/falcosecurity/libs/pull/783

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Review by FedeDP — APPROVED** [review]
/approve

File: driver/ppm_fillers.c [hunk 1]
```diff
@@ -2409,6 +2434,26 @@ int f_sys_recv_x(struct event_filler_arguments *args)
...
 }
 
+int f_sys_recvfrom_e(struct event_filler_arguments *args)
+{
+	int res = 0;
+	unsigned long val = 0;
+	s32 fd = 0;
+
+	/* Parameter 1: fd (type: PT_FD) */
+	syscall_get_arguments_deprecated(current, args->regs, 0, 1, &val);
```
Comment (hbrueckner) [inline driver/ppm_fillers.c:2444]: I think this should consider `socketcall` here:

```suggestion
	if(!args->is_socketcall)
	{
		syscall_get_arguments_deprecated(current, args->regs, 0, 1, &val);
	}
#ifndef UDIG
	else
	{
		val = args->socketcall_args[0];
	}
#endif

```

File: driver/ppm_fillers.c [hunk 2]
```diff
@@ -2409,6 +2434,26 @@ int f_sys_recv_x(struct event_filler_arguments *args)
...
+	fd = (s32)val;
+	res = val_to_ring(args, (s64)fd, 0, false, 0);
+	CHECK_RES(res);
+
+	/* Parameter 2: size (type: PT_UINT32) */
+	syscall_get_arguments_deprecated(current, args->regs, 2, 1, &val);
```
Comment (hbrueckner) [inline driver/ppm_fillers.c:2450]: same here:
```suggestion
	if(!args->is_socketcall)
	{
		syscall_get_arguments_deprecated(current, args->regs, 2, 1, &val);
	}
#ifndef UDIG
	else
	{
		val = args->socketcall_args[2];
	}
#endif
```

File: driver/ppm_fillers.c [hunk 3]
```diff
@@ -2712,6 +2757,21 @@ int f_sys_sendmsg_x(struct event_filler_arguments *args)
...
 }
 
+int f_sys_recvmsg_e(struct event_filler_arguments *args)
+{
+	unsigned long val = 0;
+	int res = 0;
+	s32 fd = 0;
+
+	/* Parameter 1: fd (type: PT_FD)*/
+	syscall_get_arguments_deprecated(current, args->regs, 0, 1, &val);
```
Comment (hbrueckner) [inline driver/ppm_fillers.c:2767]: here as well:
```suggestion
	if(!args->is_socketcall)
	{
		syscall_get_arguments_deprecated(current, args->regs, 0, 1, &val);
	}
#ifndef UDIG
	else
	{
		val = args->socketcall_args[0];
	}
#endif
```

**Review by hbrueckner — COMMENTED** [review]
Hi @Andreagit97 , except for the proposed `socketcall` changes (looks like being focussed on that today), this looks good to me.

File: driver/ppm_fillers.c (hunk 1 above)
Comment (Andreagit97) [reply driver/ppm_fillers.c]: great catch ty :)

**Review by FedeDP — APPROVED** [review]
/approve

**Pull request merged**